src/gha_runner/clouddeployment.py

```diff
@@ -0,0 +1,235 @@
+        -------
+        dict
+            A dictionary of parameters for the AWS API call.
+
+        """
+        params = {
+            "ImageId": self.image_id,
+            "InstanceType": self.instance_type,
+            "MinCount": count,
+            "MaxCount": count,
+            "TagSpecifications": self.tags,
+            "UserData": self.__build_user_data(**user_data_params),
+        }
+        if self.subnet_id != "":
+            params["SubnetId"] = self.subnet_id
+        if self.security_group_id != "":
+            params["SecurityGroupIds"] = [self.security_group_id]
+        if self.iam_role != "":
+            params["IamInstanceProfile"] = {"Name": self.iam_role}
+        return params
+
+    def create_instance(self, count: int) -> list[str]:
+        ec2 = boto3.client("ec2", region_name=self.region_name)
+        userDataParams = {
+            "token": self.gh_runner_token,
+            "repo": self.repo,
+            "homedir": self.home_dir,
+            "script": self.script,
+            "runner_release": self.runner_release,
+            "labels": self.labels,
+        }
+        params = self.__build_aws_params(count, userDataParams)
+        result = ec2.run_instances(**params)
+        instances = result["Instances"]
+        ids = [instance["InstanceId"] for instance in instances]
+        return ids
+
+    def remove_instances(self, ids: list[str]):
+        ec2 = boto3.client("ec2", self.region_name)
+        params = {
+            "InstanceIds": ids,
+        }
+        ec2.terminate_instances(**params)
+
+    def wait_until_ready(self, ids: list[str], **kwargs):
+        ec2 = boto3.client("ec2", self.region_name)
+        waiter = ec2.get_waiter("instance_running")
+        # Pass custom config for the waiter
+        if kwargs:
+            waiter.wait(InstanceIds=ids, WaiterConfig=kwargs)
+        # Otherwise, use the default config
+        else:
+            waiter.wait(InstanceIds=ids)
+
+    def wait_until_removed(self, ids: list[str], **kwargs):
+        ec2 = boto3.client("ec2", self.region_name)
+        waiter = ec2.get_waiter("instance_terminated")
+        waiter.wait(InstanceIds=ids)
+        if kwargs:
+            waiter.wait(InstanceIds=ids, WaiterConfig=kwargs)
+        else:
+            waiter.wait(InstanceIds=ids)
+
+    def instance_running(self, id: str) -> bool:
+        ec2 = boto3.client("ec2", self.region_name)
+        params = {
+            "InstanceIds": [id],
+        }
+        response = ec2.describe_instances(**params)
+        # Loop through the response to find the state of the instance
+        for reservation in response["Reservations"]:
+            for instance in reservation["Instances"]:
+                instance_state = instance["State"]["Name"]
+                if instance_state == "running":
+                    return True
+                else:
+                    return False
+        return False
+
+    def __build_user_data(self, **kwargs) -> str:
```

Comment:
As above, avoid name-mangling unless there's a really good need for it.

Reply:
Fixed.